### ISSUE TITLE
When one component is used several times in a partial view , The component properties uses same context

### ISSUE THREAD

**Issue by GitSahib** [emberjs/ember.js#15645]
`File my-component.js
export default Ember.Component.extend({
    //filter_criteria
    filter_criteria: {sourceSystem:'',aiType:'',status:''},
   sourceSystem:Ember.A(['a','b','c']),
   aiType.A(['a','b','c']),
   status.A(['a','b','c']),
  items: Ember.computed('data', function(){
    var action_items = this.get('data');
    var filter_criteria = this.get('filter_criteria');
    //filter actions. 
    var filtered_items = action_items.filter(function(action){
        //get all filter criteria keys
        var criteria_keys = Object.keys(filter_criteria);
        //filter this action by key if criteria value is not the same or empty exclude this action
        for (var i = criteria_keys.length - 1; i >= 0; i--) {
            let key = criteria_keys[i];
            if(filter_criteria[key] && action[key]){
                let search  = filter_criteria[key].toUpperCase(),// search value
                content     = action[key].toUpperCase();//search content
                var found   = content.indexOf(search) > -1;//search value contains in content?
             
             if(!found) {return false;}//not ? then break the loop return false will exclude this item from list
            }
        }            
        return true;//inlcude this item in the list
    });
    return filtered_items;
}),
})`
`
File my-component.hbs
          <label>Filter Items:</label>
                    <ul class="action-item-filters list-style-none" aria-labelledby="filter-actions">
                        <li>
                            <label>Status:</label>
                            {{select-list
                                value=filter_criteria.status
                                content=status
                                action=(action (mut filter_criteria.status ))
                                classBinding=":form-control filter_select"
                            }}
                        </li>
                        <li>
                            <label>Source System:</label>
                            {{select-list
                                value=filter_criteria.sourceSystem
                                content=sourceSystem
                                action=(action (mut filter_criteria.sourceSystem ))
                                classBinding=":form-control"
                            }}
                        </li>
                        <li>
                            <label>Ai Type:</label>                           
                            {{select-list                                
                                value=filter_criteria.aiType
                                content=aiType
                                action=(action (mut filter_criteria.aiType ))
                                classBinding=":form-control"
                                disable=false
                          }}
                   </li>
       </ul>
{{#each items as |item index|}}
    <div class="row">
        <div class="col-md-4">
            <p>{{item.aiName}}</p>
        </div>
        <div class="col-md-2">
            <p>{{item.status}}</p>
        </div>
        <div class="col-md-3">
            <p>{{item.sourceSystem}}</p>
        </div>
    </div>
{{/each}}`
`
File PartialView.hbs
{{my-component data=items}}
{{my-component data=items}}
{{my-component data=items}}
`
Now if I filter one of the components the filter is applied to all. 
I have found a work around to it by putting 
`this.set('filter_criteria',{
                sourceSystem:'',
                aiType:'',
                status:''
            });`
In the Init method of my-component. 

But why is it not using its own context without doing this hack?

**Comment by pixelhandler** [comment]
@GitSahib the format in the issue description is kind of hard to follow. Would you mind creating an example using http://ember-twiddle.com ? That would really help others investigate your issue.

Also, I don't think this is an ember bug, so far it sounds more of a Question. With a reproducible example, e.g. ember-twiddle, you will get more help. Also feel free to reach out to the ember community in the embercommunity.slack.com #-help channel. Get an invite to the slack channel here: https://ember-community-slackin.herokuapp.com/

**Comment by Serabe** [comment]
@GitSahib, it is explained in the guides [Object Model > Classes and Instances > Initializing instances](https://guides.emberjs.com/v2.15.0/object-model/classes-and-instances/#toc_initializing-instances). This behaviour is the same as classical prototypes.

Closing as a non-bug.

Thank you!

**Comment by GitSahib** [comment]
Here is the link for gist

https://ember-twiddle.com/2df6489a5e97e8a76e215f01af7101d3

I have commented out a line in init() method in my-component. 
When you uncomment the issue go away otherwise one filter fires the other too.

**Comment by Serabe** [comment]
@GitSahib please, read the link in my comment to go to the section of the guides where it is explained.

**Comment by GitSahib** [comment]
I am putting the same comment again. Please look at the gist link. This is not related totally with the link you provided. 

Here is the link for gist

https://ember-twiddle.com/2df6489a5e97e8a76e215f01af7101d3

I have commented out a line in init() method in my-component.
When you uncomment the issue go away otherwise one filter fires the other too.

**Comment by GitSahib** [comment]
I have put the object creation code you provided in the link. The behavior is still the same. My question is why Ember is duplicating the component data ? When I try to change the filter in one component it is changed in the other as well.

**Comment by karthiick** [comment]
https://guides.emberjs.com/v2.15.0/object-model/classes-and-instances/#toc_initializing-instances

>Arrays and objects defined directly on any Ember.Object are shared across all instances of that class. To avoid this behavior, it is encouraged to initialize those arrays and object properties during init(). Doing so ensures each instance will be unique.

@GitSahib As mentioned in the doc, array should be initialized in the `init()`, otherwise this will be shared between instances.

> When you uncomment the issue go away otherwise one filter fires the other too.
Yes, its bcoz you are initializing filterCriteria in `init`, so its not shared between instances. Hope this helps.

**Comment by GitSahib** [comment]
Thank you karthiick. 
I got it.
